src/SFA.DAS.ApplyService.Web/Controllers/Roatp/GetHelpController.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using SFA.DAS.ApplyService.Application.Apply.Roatp;
+using SFA.DAS.ApplyService.Domain.Roatp;
+using SFA.DAS.ApplyService.EmailService;
+using SFA.DAS.ApplyService.Session;
+using SFA.DAS.ApplyService.Web.Configuration;
+using SFA.DAS.ApplyService.Web.Infrastructure;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApplyService.Web.Controllers.Roatp
+{
+    [Authorize]
+    public class GetHelpController : Controller
+    {
+        private readonly ILogger<GetHelpController> _logger;
+        private readonly IQnaApiClient _qnaApiClient;
+        private readonly IApplicationApiClient _applicationApiClient;
+        private readonly IUsersApiClient _usersApiClient;
+        private readonly List<TaskListConfiguration> _taskListConfiguration;
+        private readonly ISessionService _sessionService;
+        private readonly IGetHelpWithQuestionEmailService _emailService;
+
+        private const string ApplicationDetailsKey = "Roatp_Application_Details";
+        private const string GetHelpSubmittedForPageKey = "Roatp_GetHelpSubmitted_{0}";
+
+        public GetHelpController(ILogger<GetHelpController> logger, IQnaApiClient qnaApiClient, IApplicationApiClient applicationApiClient,
+            IUsersApiClient usersApiClient, ISessionService sessionService, IOptions<List<TaskListConfiguration>> taskListConfiguration,
+            IGetHelpWithQuestionEmailService emailService)
+        {
+            _logger = logger;
+            _qnaApiClient = qnaApiClient;
+            _applicationApiClient = applicationApiClient;
+            _usersApiClient = usersApiClient;
+            _sessionService = sessionService;
+            _taskListConfiguration = taskListConfiguration.Value;
+            _emailService = emailService;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Index(Guid? applicationId, int sequenceId, int sectionId, string pageId, string title, string getHelp, string controller, string action)
+        {
+            var getHelpQuery = new GetHelpWithQuestion();
+
+            if (applicationId.HasValue && applicationId.Value != Guid.Empty)
+            {
+                var sequences = await _qnaApiClient.GetSequences(applicationId.Value);
+                var currentSequence = sequences.Single(x => x.SequenceId == sequenceId);
+                var sections = await _qnaApiClient.GetSections(applicationId.Value, currentSequence.Id);
+                var currentSection = sections.FirstOrDefault(x => x.SectionId == sectionId);
+
+                var page = await _qnaApiClient.GetPage(applicationId.Value, currentSection.Id, pageId);
```

Comment:
You could cut this down by using _qnaApiClient.GetPageBySectionNo

Reply:
Ok, will raise a tech debt story to implement that in the QnA Api Client and replace occurrences